### ISSUE TITLE
Ideas of what options to put in a window for Foundry dedicated to miscellaneous options.

### ISSUE THREAD

**Issue by Dariosky-01** [mchlnix/SMB3-Foundry#95]
My idea is simple, there may exist these options may now I am going to write them all in the miscellaneous options menu, where inside there are many or few options related to that change, and these divided by tab, which are separated vertically or horizontally, this would ultimately depend on whether it was more convenient to view one way rather than another, most likely, the tabs would be more convenient to view horizontally such as the choice of levels to be modified to be chosen within the maps visible in Foundry.

I think we could think about creating them, however, it is logical to think that for some of them we could think immediately because they are easy to add and probably others I will say maybe later in time, after the hard work we would have to do to add them.

It goes without saying that if a modification is easy to add, anyone could immediately enjoy the benefit of modifying this software.

Now I will make a list but I immediately start by saying that those currently present that can be observed on Workshop would be added to this list I will do, however I hope that, if possible, it is all done in a more understandable way possibly because the only sin for me has this software is to be too "synthetic", too far from a more "human" way or if you want direct in the approach with the person who will use it, in fact there in this menu you see code values ​​as if we were editing with hex editor for how they look there, so do you think that in Foundry they should appear simpler to the point that it should not be necessary to be aware of more technical knowledge such as what a value means, this would help a lot those who are less aware on so many topics, a very trivial example would be that of the temporal speed in a modification there for the duration of the super power, it would have been more understandable to modify via an instrument with something that made the measurement understand visibly as well as with numbers or hex values, with a horizontal or vertical linear potentiometer.

Now in the list I am about to do I might accidentally talk about some option that Workshop has but if I will it is because I do not remember if they exist there or not, but it is definitely worth reading because every idea I express it in a way that lends itself to be much simpler than the way Workshop presents it, I often write something more as well so read everything.

One option I would like to see is to give me a way to change the color of the inventory bar at the bottom of the map, having the opportunity to change the external colors from time to time I mean the edges and the internal color. The colors to change should be those of the inventory in its closed resting state, those of its interior when it is opened, and the interior color when it turns, because even if I could change for the pink interior where the clothes earned then the inside of the inventory open, and outside inside it had a different color like deep blue, when it turns I would always see the inside a pink when turning it would tend to return closed, even if it had the inside color of the 'open inventory of another color example green and closed inventory, as I wrote before it was blue, and this must be changed, it must be added as a particular modification for the external bar, something foreign therefore to the colors of the status bars inside the levels.

As another set in another card the coloring of Mario in all its transformations should be added and even when he is small, matching the colors of these together with those are returned at the exit of the level, because they are different.
Anyway, nobody should go to use the emulator to test it, it would be enough to put the image of mario and luigi still big and small like a photo in the window to look at the colors that they would be testing in real time.
Besides, remember to add the same colors to Mario and Luigi's drawings in the current world notification windows in all their possible combinations of clothes they could wear.

A similar modification to add on the same window or other tab could be to change the colors of the princess and bowser including the colors of them in the letters.

Unused palettes should be made possible
to separate those of the worlds
1 and 3. Someone told me, but I don't remember for sure the details, that there are palettes not to use, one is sure, I could be wrong but it could be two, so you could create a set to match palettes in the worlds, so that you they can choose whether to have a different one for each one or some in common as in the original it already exists with world 1 and 3

You could create a set in another tab to place the darkness of the third screen of world 8 where you want, always keeping the fact that it is on the third screen so you can only choose between world 3,6 and 8.
If it were easy to change the hex code to add more options to make dark in the map work anywhere according to set within this tab, it would also be very good, obviously making windowed settings available to activate dark working in multiple screens of a chosen world would be good, because this I could use for every world including world 9 and then more than anything else just to make the game cuter more than for a purpose in a world like 9.
I am thinking of this set to use it to my advantage to make the player's life difficult along with other sets that I will say here.

The sprites circumscribe Mario in the dark to show a small part of the map they could be removed with a set of your choice in that previous card this would open the circle by showing Mario inside an open map square, this would be good because sometimes they excessively disturb the sprites approaching the edges of the map, they create an undesirable effect, unless you look for a simple solution to change this defect, but I think you can't, we need to study the system better to see if the solution to this problem is feasible and that however, a solution gives an easy way, together with the other options, to make everything work without many problems.

You could in a tab give way to remove the "current world notification" window, to speed up the game or just because you don't like it, but I hope that the effect with the stars Mario creates on the map at the beginning of the game is not removed.
Always in the same card, possibly tighten with the number of cards,
you could give the option to remove the title of world 9 or always in the same window write the title to be displayed, perhaps displaying it in real time in the same window, the same thing for the internal writing of the windows "notification current world".

Another option in another tab could allocate multiple locks meaning the blocks connected to the castles.
You could use up to 11, which is the maximum number, available in the game even for one world only, currently the maximum number of blocks that can be placed in a world, with a boulder of 11 for the whole game, is 4, I don't like this, I would like more choice.
I don't know if another idea was possible, but if it is possible to really only have 11 for the whole game then the setting in this tab could only give way to allocate this closed number of 11 blocks of castles, for the worlds perhaps always with a number maximum of 11 per world, surely it would not give map editor compatible in case, but if one day there was an internal Foundry map editor with those same functions, this could solve the possible problem and make the use of Map Editor obsolete.

Another tab could give way to block the lives of Mario and Luigi and together with this, remove the number of lives in the inventory bar by setting an empty spite therefore in the same color as the bar, and together with this screw lock function, and the same in the world notifications windows speaking for the number of lives.

Another card could be dedicated is the one for the "game genie" to be set in the game, but so that the system, once modified, is reversible, if it can't be done I don't think it's a good idea, the "game genie". However they should not be codes to be written manually, but according to my idea they should be codes already established in the software, codes where in the window you could choose which ones to add, or adjust according to your needs to modify in a very personalized way, if it was not possible this extra detail, however, would not be important, but it would certainly be a shame.

Finally, however, I would like to point out that it could very well make a card that unites a series of simple settings, probably also some of those mentioned above and that can easily enter a card where for example another setting could be
in addition, one that can change the total speed of the game, sometimes it seems a bit slow to me and I would like to be able to play with a slightly higher speed, if I say this I do not say it at random, I say it in order not to possibly set the speed in the emulator.
(always if it is possible to do it)
if possible only change the game speed of Mario or the enemies but all together without changing the speed of the music.

There are however two other settings that I published as ideas recently in a publication, my first idea, where I present a particular setting idea should be added and which I mean in this window in a tab.

There I also recommend to give way to change the music of the world by choosing among those available the game has, so if you wanted you could also set an equal song for more worlds, in case you want this for more worlds.

However, I highly recommend going to see that publication because it contains this setting I was talking about which is very particular, the explanation is long there for this reason I have dedicated a publication to it, adding it later in the miscellaneous options window and in a dedicated tab would be very easy , the problem is to understand the operation as I imagined it in the smallest details.

I remember once again that along with these settings there should also be those visible on Workshop miscellaneous options, like that specific option that dictates what the princess letters should give me that would add to my inventory.

**Comment by TheJoeSmo** [comment]
I do not believe we need a miscellaneous tab in Foundry.  I find it vague and overall unhelpful in development of a ROM.  For the points you mentioned (such as more palette editing, patches, and locks), they warrant their own issues.  You are making a solution to a problem you created; you are creating a tab that is otherwise not needed.  For this reason, I disagree with your issue on the basis that a miscellaneous tab is not something I believe Foundry needs at this point.  Feel free to write the palette editing and other points as separate issues.  If we find it needed after doing other issues than an issue just to make the misc. tab can be created.

**Comment by Dariosky-01** [comment]
I just wanted to accumulate a series of novelties all to be selected in a window, novelties of things that a large part of Workshop has and other things that would be missing from the same Workshop and others that would help on other functional fronts of how, for example, it should scrolling of the worlds, in my idea world 7 is not said to be the first or last or the same world 7 as in the original, but it is related to the setting in a tab of this window, so I would say that this window has a wide range of changes and all concentrated there for ease of use, if then this idea, as I thought it is not very functional is another, but certainly it is very useful to have many options mentioned in this publication, they allow many small things that otherwise for how Foundry is currently structured could not be done, for now on this point of view it is highly crude, for example I cannot decide which weapons will give me the letter of the princess, the which is very bad, but this can be done with Workshop, and here's my problem what it is, to make it similar, but similar because it is better and not worse with less functionality, the goal here is to improve Foundry.

**Comment by mchlnix** [comment]
- [ ] change score bar color (isn't this part of some color palette)
- [ ] change colors of Mario and Luigi in their power up states (could be complicated)
- [ ] change color of the Princess and Bowser (a bit arbitrary)
- [ ] Change Palette of the Overworld? (This is possible in Scribe)
- [ ] Darkness Effect of World 8 Screen 3 everywhere
- [ ] Change overworld music (I think already done in Scribe)

That Overworlds only have 4 possible locks is hard coded into the game. The same with the lives and world notification changes.

In general I wouldn't want to make anything available in the editor, that would need the editor to patch the ROM, since this is not reversible or easily recognizable.

What if someone "patches" their rom, using the editor, then reopens it and wants to undo that action? I guess it could be made obvious, that it is a one way street, but we already have patches for that.
